numpyro/distributions/util.py

```diff
@@ -290,6 +292,11 @@ def promote_shapes(*args, shape=()):
                 if len(s) < num_dims else arg for arg, s in zip(args, shapes)]
 
 
+def get_dtypes(*args):
+    return [canonicalize_dtype(type(arg)) if isinstance(arg, Number)
+            else canonicalize_dtype(onp.dtype(arg)) for arg in args]
```

Comment:
I am still not familiar with how jax manages dtypes. Somewhere I see `lax._dtype`, somewhere I see `canonicalize_dtype`. >"<

Reply:
@fehiepsi - Based on my limited exploration, my conclusion was that `canonicalize_dtype` will take care of default conversions, but can't handle python scalars. I just realized that this function can be simplified to `canonicalize_dtype(onp.result_type(arg))` (will do that in a follow up PR). `lax._dtype` just delegates to `onp.result_type`.

Reply:
Thanks for explaining!